src/services/Store/DataManager/DataProvider.tsx

```diff
@@ -36,98 +22,42 @@ interface EncryptedStorage {
   setEncryptedCache(ls: string): void;
   destroyEncryptedCache(): void;
   setUnlockPassword(pwd: string): void;
-  getUnlockPassword(): string;
 }
 export type IDataContext = DataCacheManager & EncryptedStorage;
 
 export const DataContext = createContext({} as IDataContext);
 
-const isSafari = () => /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
-
 export const DataProvider: React.FC = ({ children }) => {
+  const currentDB = useMemo(() => getCurrentDBConfig(), []);
+
+  const dispatch = useDispatch();
+  const legacyState = useSelector(getAppState);
+
+  const resetAppDb = (newDb = initialLegacyState) => {
+    // resetDb(newDb); // Reset the persistence layer
+    dispatch({
+      type: 'RESET',
+      payload: { data: newDb }
+    }); // Reset the Context
```

Comment:
Can you use an action creator here?

Reply:
We want to remove the legacy reducer as soon as this PR is merged.
The action declaration will change at the moment, and I prefer to handle the typing then. 
Added a `todo` comment as a reminder.